database/src/main/scala/cromwell/database/SqlDatabase.scala

```diff
@@ -178,22 +178,12 @@ trait SqlDatabase extends AutoCloseable {
                                                   (implicit ec: ExecutionContext):
   Future[Traversable[(Execution, ExecutionInfo)]]
 
-  protected def addMetadataEvent(workflowUuid: String,
-                                 key: String,
-                                 value: String,
-                                 valueType: String,
-                                 timestamp: Timestamp)
-                                (implicit ec: ExecutionContext): Future[Unit]
-
-  protected def addMetadataEvent(workflowUuid: String,
-                                 key: String,
-                                 callFqn: String,
-                                 index: Option[Int],
-                                 attempt: Int,
-                                 value: String,
-                                 valueType: String,
-                                 timestamp: Timestamp)
-                                (implicit ec: ExecutionContext): Future[Unit]
+  /**
+    * Add metadata events to the database transactionally. normalized type structure is as follows:
+    * (WorkflowId, MetadataKey, Option[CallFqn, CallIndex, CallAttempt], MetadataValue, MetadataValueType, Timestamp)
+    */
+  protected def addMetadata(events: Iterable[(String, String, Option[(String, Option[Int], Int)], String, String, Timestamp)])
+                           (implicit ec: ExecutionContext): Future[Unit]
```

Comment:
@kshakir You do touch on the one issue i see w/ case class type wrappers, the attractive nuisance they provide for jamming all sorts of logic in there instead of just being a typed wrapper